### PR TITLE
Only resolve inline script specifiers in the static build

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- The 5c98456028df94aa2edef62512b24e6936d4c0b8 commit is the new code here.
- The other commit is a revert of a revert. Had to revert the new static build enhancements due to it breaking the docs site.
- In order to not rely on vite HTML post-processing the static build resolves inline script specifiers by mapping them to the hashed built file. This doesn't work for the legacy build as Vite-HTML doesn't use the same underlying API. We will want to ask Vite to expose their API, but for now since it's not needed by the legacy build, we can just not resolve the specifiers at render-time.

### CODE REVIEW

File: packages/astro/src/core/build/static-build.ts [hunk 1]
```diff
@@ -151,14 +203,36 @@ async function generatePath(path: string, opts: StaticBuildOptions, gopts: Gener
 			route: pageData.route,
 			routeCache,
 			logging,
-			pathname: path,
+			pathname,
```
Comment (jonathantneal) [inline packages/astro/src/core/build/static-build.ts:206]: On L195 I was like "huh" and on L206 I’m like "ahh". 👍

File: packages/astro/src/vite-plugin-markdown/index.ts [hunk 2]
```diff
@@ -51,8 +51,12 @@ ${setup}`.trim();
 					astroResult = `${prelude}\n${astroResult}`;
 				}
 
+				const filenameURL = new URL(`file://${id}`);
+				const pathname = filenameURL.pathname.substr(config.projectRoot.pathname.length - 1);
```
Comment (jonathantneal) [inline packages/astro/src/vite-plugin-markdown/index.ts:55]: Nit Idea: Is `filenameURL` being used to sanitize the URL? If so, I think a future PR should introduce posix sanitation methods. Then this line could be like:
```js
/** Pathname, with a leading slash. */
const pathname = posix.path(id).slice(config.projectRoot.pathname.length - 1)
```

**Review by jonathantneal — APPROVED** [review]
Generally, I follow. 👍

File: packages/astro/src/vite-plugin-markdown/index.ts (hunk 2 above)
Comment (matthewp) [reply packages/astro/src/vite-plugin-markdown/index.ts:55]: It's being used to turn it into the pathname instead yeah. Yeah that way would presumably work too.

**Pull request merged**